// @adonsio/adn-tournament/bundles/adonsio-adn-tournament.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common')) :
    typeof define === 'function' && define.amd ? define('@adonsio/adn-tournament', ['exports', '@angular/core', '@angular/common'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.adonsio = global.adonsio || {}, global.adonsio['adn-tournament'] = {}), global.ng.core, global.ng.common));
}(this, (function (exports, core, common) { 'use strict';

    var DoubleEliminationTreeComponent = /** @class */ (function () {
        function DoubleEliminationTreeComponent() {
        }
        DoubleEliminationTreeComponent.prototype.ngOnChanges = function (changes) {
            this.losersBracket = this.tournament.rounds.filter(function (round) {
                return round.type === 'Loserbracket';
            });
            this.winnersBracket = this.tournament.rounds.filter(function (round) {
                return round.type === 'Winnerbracket';
            });
            //@ts-ignore
            this.final = this.tournament.rounds.filter(function (round) {
                return round.type === 'Final';
            }).shift();
        };
        return DoubleEliminationTreeComponent;
    }());
    DoubleEliminationTreeComponent.decorators = [
        { type: core.Component, args: [{
                    selector: 'adn-double-elimination-tree',
                    template: "<div class=\"ngtt-de__container\">\r\n    <div class=\"ngtt-de__winners-bracket-wrapper\">\r\n      <ng-container *ngFor=\"let winnersround of winnersBracket\">\r\n        <!--      {{winnersround.round}}-->\r\n        <div class=\"ngtt-de__winners-round-wrapper\">\r\n          <ng-container *ngFor=\"let match of winnersround.matches\">\r\n            <div class=\"ngtt-de__winners-match-wrapper\">\r\n              <div class=\"ngtt-de__match-component-container\">\r\n                <ng-container *ngTemplateOutlet=\"matchTemplate; context:{$implicit: match}\"></ng-container>\r\n              </div>\r\n            </div>\r\n          </ng-container>\r\n        </div>\r\n      </ng-container>\r\n      <div class=\"ngtt-de__winners-round-wrapper\">\r\n        <div class=\"ngtt-de__finale-match-wrapper\">\r\n          <div class=\"ngtt-de__match-component-container\">\r\n            <ng-container *ngTemplateOutlet=\"matchTemplate; context:{$implicit: final.matches[0]}\"></ng-container>\r\n          </div>\r\n        </div>\r\n      </div>\r\n    </div>\r\n    <div class=\"ngtt-de__losers-bracket-wrapper\">\r\n      <ng-container *ngFor=\"let losersround of losersBracket\">\r\n        <!--      {{losersround.round}}<br>-->\r\n        <div class=\"ngtt-de__losers-round-wrapper\">\r\n          <ng-container *ngFor=\"let match of losersround.matches\">\r\n            <div class=\"ngtt-de__losers-match-wrapper\">\r\n              <div class=\"ngtt-de__match-component-container\">\r\n                <ng-container *ngTemplateOutlet=\"matchTemplate; context:{$implicit: match}\"></ng-container>\r\n              </div>\r\n            </div>\r\n          </ng-container>\r\n        </div>\r\n      </ng-container>\r\n    </div>\r\n  </div>\r\n  ",
                    styles: [""]
                },] }
    ];
    DoubleEliminationTreeComponent.ctorParameters = function () { return []; };
    DoubleEliminationTreeComponent.propDecorators = {
        matchTemplate: [{ type: core.Input }],
        tournament: [{ type: core.Input }]
    };

    var components$1 = [DoubleEliminationTreeComponent];
    var AdnDoubleEliminationTreeModule = /** @class */ (function () {
        function AdnDoubleEliminationTreeModule() {
        }
        return AdnDoubleEliminationTreeModule;
    }());
    AdnDoubleEliminationTreeModule.decorators = [
        { type: core.NgModule, args: [{
                    declarations: components$1,
                    imports: [
                        common.CommonModule
                    ],
                    exports: components$1
                },] }
    ];

    var SingleEliminationTreeComponent = /** @class */ (function () {
        function SingleEliminationTreeComponent() {
        }
        SingleEliminationTreeComponent.prototype.ngOnInit = function () {
        };
        SingleEliminationTreeComponent.prototype.ngOnChanges = function (changes) {
            if (changes.hasOwnProperty('tournament') && changes.tournament.currentValue) {
                this.winnersBracket = this.tournament.rounds.filter(function (round) {
                    return round.type === 'Winnerbracket';
                });
            }
            console.log(this.winnersBracket);
            //@ts-ignore
            this.final = this.tournament.rounds.filter(function (round) {
                return round.type === 'Final';
            }).shift();
            console.log(this.final);
        };
        return SingleEliminationTreeComponent;
    }());
    SingleEliminationTreeComponent.decorators = [
        { type: core.Component, args: [{
                    selector: 'adn-single-elimination-tree',
                    template: "<div class=\"ngtt-se__container\">\r\n    <div class=\"ngtt-se__bracket-wrapper\">\r\n      <ng-container *ngFor=\"let winnersround of winnersBracket; let i = index\">\r\n        <div class=\"ngtt-se__round-wrapper\">\r\n          <ng-container *ngFor=\"let match of winnersround.matches\">\r\n            <div class=\"ngtt-se__match-wrapper\">\r\n              <div class=\"ngtt-se__match-component-container\">\r\n                <ng-container *ngTemplateOutlet=\"matchTemplate; context:{$implicit: match}\"></ng-container>\r\n              </div>\r\n            </div>\r\n          </ng-container>\r\n        </div>\r\n      </ng-container>\r\n      <ng-container>\r\n        <div class=\"ngtt-se__round-wrapper\">\r\n          <div class=\"ngtt-se__match-wrapper\">\r\n            <div class=\"ngtt-se__finale-match-wrapper\">\r\n              <div class=\"ngtt-se__finale-component-container\">\r\n                <ng-container\r\n                  *ngTemplateOutlet=\"matchTemplate; context:{$implicit: final.matches[0]}\"></ng-container>\r\n              </div>\r\n              <div class=\"ngtt-se__match-component-container\" *ngIf=\"final.matches.length > 1\">\r\n                <ng-container *ngTemplateOutlet=\"matchTemplate; context:{$implicit: final.matches[1]}\"></ng-container>\r\n              </div>\r\n            </div>\r\n          </div>\r\n        </div>\r\n      </ng-container>\r\n    </div>\r\n  </div>\r\n  ",
                    styles: [""]
                },] }
    ];
    SingleEliminationTreeComponent.ctorParameters = function () { return []; };
    SingleEliminationTreeComponent.propDecorators = {
        matchTemplate: [{ type: core.Input }],
        tournament: [{ type: core.Input }]
    };

    var components = [SingleEliminationTreeComponent];
    var AdnSingleEliminationTreeModule = /** @class */ (function () {
        function AdnSingleEliminationTreeModule() {
        }
        return AdnSingleEliminationTreeModule;
    }());
    AdnSingleEliminationTreeModule.decorators = [
        { type: core.NgModule, args: [{
                    declarations: components,
                    imports: [
                        common.CommonModule
                    ],
                    exports: components
                },] }
    ];

    var modules = [AdnSingleEliminationTreeModule, AdnDoubleEliminationTreeModule];
    var AdnTournamentTreeModule = /** @class */ (function () {
        function AdnTournamentTreeModule() {
        }
        return AdnTournamentTreeModule;
    }());
    AdnTournamentTreeModule.decorators = [
        { type: core.NgModule, args: [{
                    declarations: [],
                    imports: modules,
                    exports: [AdnSingleEliminationTreeModule, AdnDoubleEliminationTreeModule]
                },] }
    ];

    /*
     * Public API Surface of adn-tournament
     */

    /**
     * Generated bundle index. Do not edit.
     */

    exports.AdnDoubleEliminationTreeModule = AdnDoubleEliminationTreeModule;
    exports.AdnSingleEliminationTreeModule = AdnSingleEliminationTreeModule;
    exports.AdnTournamentTreeModule = AdnTournamentTreeModule;
    exports.ɵa = SingleEliminationTreeComponent;
    exports.ɵb = DoubleEliminationTreeComponent;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
