// @adonsio/adn-tournament/esm2015/lib/single-elimination-tree/single-elimination-tree.component.js
import { Component, Input } from '@angular/core';
export class SingleEliminationTreeComponent {
    constructor() {
    }
    ngOnInit() {
    }
    ngOnChanges(changes) {
        if (changes.hasOwnProperty('tournament') && changes.tournament.currentValue) {
            this.winnersBracket = this.tournament.rounds.filter(round => {
                return round.type === 'Winnerbracket';
            });
        }
        console.log(this.winnersBracket);
        //@ts-ignore
        this.final = this.tournament.rounds.filter(round => {
            return round.type === 'Final';
        }).shift();
        console.log(this.final);
    }
}
SingleEliminationTreeComponent.decorators = [
    { type: Component, args: [{
                selector: 'adn-single-elimination-tree',
                template: "<div class=\"ngtt-se__container\">\r\n    <div class=\"ngtt-se__bracket-wrapper\">\r\n      <ng-container *ngFor=\"let winnersround of winnersBracket; let i = index\">\r\n        <div class=\"ngtt-se__round-wrapper\">\r\n          <ng-container *ngFor=\"let match of winnersround.matches\">\r\n            <div class=\"ngtt-se__match-wrapper\">\r\n              <div class=\"ngtt-se__match-component-container\">\r\n                <ng-container *ngTemplateOutlet=\"matchTemplate; context:{$implicit: match}\"></ng-container>\r\n              </div>\r\n            </div>\r\n          </ng-container>\r\n        </div>\r\n      </ng-container>\r\n      <ng-container>\r\n        <div class=\"ngtt-se__round-wrapper\">\r\n          <div class=\"ngtt-se__match-wrapper\">\r\n            <div class=\"ngtt-se__finale-match-wrapper\">\r\n              <div class=\"ngtt-se__finale-component-container\">\r\n                <ng-container\r\n                  *ngTemplateOutlet=\"matchTemplate; context:{$implicit: final.matches[0]}\"></ng-container>\r\n              </div>\r\n              <div class=\"ngtt-se__match-component-container\" *ngIf=\"final.matches.length > 1\">\r\n                <ng-container *ngTemplateOutlet=\"matchTemplate; context:{$implicit: final.matches[1]}\"></ng-container>\r\n              </div>\r\n            </div>\r\n          </div>\r\n        </div>\r\n      </ng-container>\r\n    </div>\r\n  </div>\r\n  ",
                styles: [""]
            },] }
];
SingleEliminationTreeComponent.ctorParameters = () => [];
SingleEliminationTreeComponent.propDecorators = {
    matchTemplate: [{ type: Input }],
    tournament: [{ type: Input }]
};
