// @adonsio/adn-tournament/esm2015/lib/double-elimination-tree/double-elimination-tree.component.js
import { Component, Input } from '@angular/core';
export class DoubleEliminationTreeComponent {
    constructor() {
    }
    ngOnChanges(changes) {
        this.losersBracket = this.tournament.rounds.filter(round => {
            return round.type === 'Loserbracket';
        });
        this.winnersBracket = this.tournament.rounds.filter(round => {
            return round.type === 'Winnerbracket';
        });
        //@ts-ignore
        this.final = this.tournament.rounds.filter(round => {
            return round.type === 'Final';
        }).shift();
    }
}
DoubleEliminationTreeComponent.decorators = [
    { type: Component, args: [{
                selector: 'adn-double-elimination-tree',
                template: "<div class=\"ngtt-de__container\">\r\n    <div class=\"ngtt-de__winners-bracket-wrapper\">\r\n      <ng-container *ngFor=\"let winnersround of winnersBracket\">\r\n        <!--      {{winnersround.round}}-->\r\n        <div class=\"ngtt-de__winners-round-wrapper\">\r\n          <ng-container *ngFor=\"let match of winnersround.matches\">\r\n            <div class=\"ngtt-de__winners-match-wrapper\">\r\n              <div class=\"ngtt-de__match-component-container\">\r\n                <ng-container *ngTemplateOutlet=\"matchTemplate; context:{$implicit: match}\"></ng-container>\r\n              </div>\r\n            </div>\r\n          </ng-container>\r\n        </div>\r\n      </ng-container>\r\n      <div class=\"ngtt-de__winners-round-wrapper\">\r\n        <div class=\"ngtt-de__finale-match-wrapper\">\r\n          <div class=\"ngtt-de__match-component-container\">\r\n            <ng-container *ngTemplateOutlet=\"matchTemplate; context:{$implicit: final.matches[0]}\"></ng-container>\r\n          </div>\r\n        </div>\r\n      </div>\r\n    </div>\r\n    <div class=\"ngtt-de__losers-bracket-wrapper\">\r\n      <ng-container *ngFor=\"let losersround of losersBracket\">\r\n        <!--      {{losersround.round}}<br>-->\r\n        <div class=\"ngtt-de__losers-round-wrapper\">\r\n          <ng-container *ngFor=\"let match of losersround.matches\">\r\n            <div class=\"ngtt-de__losers-match-wrapper\">\r\n              <div class=\"ngtt-de__match-component-container\">\r\n                <ng-container *ngTemplateOutlet=\"matchTemplate; context:{$implicit: match}\"></ng-container>\r\n              </div>\r\n            </div>\r\n          </ng-container>\r\n        </div>\r\n      </ng-container>\r\n    </div>\r\n  </div>\r\n  ",
                styles: [""]
            },] }
];
DoubleEliminationTreeComponent.ctorParameters = () => [];
DoubleEliminationTreeComponent.propDecorators = {
    matchTemplate: [{ type: Input }],
    tournament: [{ type: Input }]
};
